server/models/questions/forkQuestion.js

```diff
@@ -0,0 +1,90 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Number} param0.questionId
+ * @return {Promise}
+ */
+
+function forkQuestion({ username, questionId }) {
+  return new Promise((resolve, reject) => {
+    pool.getConnection((err, connection) => {
+      if (err) {
+        return reject(err)
+      }
+      connection.beginTransaction((err) => {
+        if (err) {
+          connection.release()
+          return reject(err)
+        }
+        connection.query(
+          `INSERT INTO questions(creator,type,name,problem_statement,input_format,
+            output_format,constraints,options,correct,difficulty)
+            SELECT creator,type,name,problem_statement,input_format,
+            output_format,constraints,options,correct,difficulty 
+            FROM questions q
+            JOIN questions_editors qe ON q.id=qe.question_id 
+            WHERE q.id=? AND qe.editor=?`,
+          [questionId, username],
+          (error, results) => {
+            if (error || !results) {
+              connection.release()
+              return reject(error)
+            }
+
+            if (!results.insertId) {
+              return reject('Invalid question_id or unauthorized user')
+            }
+
+            const forkedQuestionId = results.insertId
+
+            connection.query(
+              `INSERT INTO questions_editors(question_id, editor) VALUES(?,?)`,
+              [forkedQuestionId, username],
+              (error) => {
+                if (error) {
```

Comment:
```suggestion
              (error, res) => {
                if (error || res === undefined) {
```

Reply:
I think this is redundant because in the above query authorization has been checked and if there is any problem it will be reflected in error

Reply:
Yeah, it would be redundant, since `res=== undefined` implies error is true and vice-versa. I don't think, there should be any case where `error` is a false value and `res` is still undefined.